packages/sdk/src/stores/google-datastore/store.ts

```diff
@@ -22,37 +25,61 @@ export default function <D>(kind: string, store: Datastore): SortedStore<string,
     });
   }
   async function has(id: string) {
-    try {
-      await get(id);
-      return true;
-    } catch (err) {
-      return false;
-    }
+    // horribly ineficient to actually query data to see if it exists, but cannot find a better way
+    return exists(await get(id));
   }
   async function del(id: string) {
     await store.delete(makeKey(id));
   }
   async function entries() {
-    return [];
+    const [results] = await store.createQuery(kind).run();
+    return results.map((result) => {
+      return [result[store.KEY].name, result];
+    }) as [string, D][];
   }
   async function values() {
-    return [];
+    const [results] = await store.createQuery(kind).run();
+    return results;
   }
+  // all this does is map over full entries to return the id of the entry. have not found a better way to query this.
   async function keys() {
-    return [];
+    const [results] = await store.createQuery(kind).run();
+    return results.map((result) => {
+      return result[store.KEY].name;
+    }) as string[];
   }
+  // theres no way to really do built into the store client. Google recommends managing a size entry yourself.
   async function size() {
-    return 0;
+    throw new Error("size not supported in google store");
+    // typescript complains if theres no return value
+    return NaN;
   }
-  // TODO: implement these functions
+  // this actually queries all values, then batches them to delete N at a time. This is a horrible way
+  // to do this, but I could not find a better way to "drop" all kinds from the table.
   async function clear() {
-    throw new Error("clear is not implemented");
+    await highland(store.createQuery(kind).runStream())
+      .map((val: any) => {
+        return val[store.KEY];
+      })
+      .batch(500)
+      .map(async (keys) => {
+        return store.delete(keys);
+      })
+      .flatMap(highland)
+      .collect()
+      .toPromise(Promise);
   }
-  async function slice() {
-    return [];
+  async function slice(id: string, length: number) {
+    const [result] = await store.createQuery(kind).filter("__key__", ">=", makeKey(id)).limit(length).run();
```

Comment:
Are we sure limit ensures that we get the _first_ `n`? As opposed to just _some_ `n` that satisfy the constraint? Unclear to me from the docs.

Reply:
its a good question, i've tried to test for this but maybe its not super precise. I basically insert a bunch of things out of order, then do a slice and check the results:
```js
    result.forEach((price: Price) => {
      assert.equal(price.identifier, "b");
      if (last == null) {
        last = price;
        return;
      }
      assert.ok(makeId(price) >= makeId(last));
    });
```

basically checking here that each result id is greater than the previous, so at the very least its ordered, though this does not check for skipped values

Reply:
ive updated the tests to check the exact values that should be returned based on the original list. things seem to check out.

Reply:
Gotcha. I think that's a good enough test.

I'm guessing this wouldn't work if we were filtering on something _other_ than key, since this probably just returns elements in the natural ordering of the db, which is by key.

LGTM!